torch_frame/data/mapper.py

```diff
@@ -129,13 +130,25 @@ def forward(
         if ser.dtype != 'object':
             raise ValueError('Multi-categorical types expect string as input')
         values = []
-        ser = ser.apply(self._row_to_tensor)
-        values = torch.tensor(sum(ser, []), device=device)
-        ser = ser.apply(lambda x: len(x))
-        offset = torch.tensor([0] + ser.tolist(), device=device)
+        original_index = ser.index
+        ser = ser.apply(self._split_by_sep)
+        ser = ser.explode()
+        ser = pd.merge(
+            ser.rename('data'),
+            self.index.rename('index'),
+            how='left',
+            left_on='data',
+            right_index=True,
+        ).dropna()
+        ser['index'] = ser['index'].astype('int64')
+        values = torch.tensor(ser['index'].values, device=device)
```

Comment:
nit:
```
values = torch.tensor(ser['index'].values, device=device, dtype=torch.long)
```

Reply:
actually, we should always use `torch.from_numpy` instead for zero-copy.

Reply:
+1